lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1585,4 +1585,6 @@ void TypeChecker::typeCheckTopLevelCodeDecl(TopLevelCodeDecl *TLCD) {
   StmtChecker(*this, TLCD).typeCheckStmt(Body);
   TLCD->setBody(Body);
   checkTopLevelErrorHandling(TLCD);
+  if (TLCD) 
```

Comment:
Since `TLCD` is dereferenced two lines above, it should be okay to drop this check.